Makefile

```diff
@@ -286,19 +286,55 @@ release-unix: clean full
 	fi
 
 #
-# make release-windows - Produces a binary release tarball containing teleport,
-# tctl, and tsh.
+# make release-windows-unsigned - Produces a binary release archive containing only tsh.
 #
-.PHONY:
-release-windows: clean all
+.PHONY: release-windows-unsigned
+release-windows-unsigned: clean all
 	@echo "---> Creating OSS release archive."
 	mkdir teleport
 	cp -rf $(BUILDDIR)/* \
 		README.md \
 		CHANGELOG.md \
 		teleport/
-	mv teleport/tsh teleport/tsh.exe
+	mv teleport/tsh teleport/tsh-unsigned.exe
 	echo $(GITTAG) > teleport/VERSION
+	zip -9 -y -r -q $(RELEASE)-unsigned.zip teleport/
+	rm -rf teleport/
+	@echo "---> Created $(RELEASE)-unsigned.zip."
+
+#
+# make release-windows - Produces an archive containing a signed release of
+# tsh.exe
+#
+.PHONY: release-windows
+release-windows: release-windows-unsigned
+	@if [ -z "$${WINDOWS_SIGNING_CERT}" ]; then \
+		echo "WINDOWS_SIGNING_CERT is unset, cannot create signed archive."; \
+		exit 1; \
+	fi
+
+	rm -rf teleport
+	@echo "---> Extracting $(RELEASE)-unsigned.zip"
+	unzip $(RELEASE)-unsigned.zip
+	
+	@echo "---> Signing Windows binary."
+	@cert_path=$$(mktemp); \
+	echo "$${WINDOWS_SIGNING_CERT}" | base64 -d > "$$cert_path" || exit 1; \
```

Comment:
I'd prefer we decode in the Drone logic, and have the Makefiles use a path to a file as input -- or rely on a wellknown (and .gitignored) path.  This keeps Make and Drone more loosely coupled. A base64 encoded file in an environment variable isn't a common build input -- we probably wouldn't use that same system with any other CI harness.

E.g. consider the case of testing locally. To me it feels obtuse to:

```console
WINDOWS_SIGNING_CERT="$(cat test.pfx | base 64)" make release-windows
```
As opposed to:

```console
WINDOWS_SIGNING_CERT=test.pfx make release-windows
```

Reply:
I don't disagree, that would be a nice improvement (and more similar to how we handle gpg for signed Linux builds).

Unfortunately `make` [immediately changes directories](https://www.gnu.org/software/make/manual/make.html#Recursion) when we use `make -C build.assets ...` so any path contained within a `$WINDOWS_SIGNING_CERT` variable will be relative to an unknown base directory by the time we try to do anything with it. `make` apparently forgets its original cwd and provides no variables or functions to recover it, let alone resolve an absolute path from a user-provide env var.

We could expect anyone testing locally to provide an absolute path but IMO that's beginning to feel similarly obtuse.

Reply:
Alright, it now uses a set file (`windows-signing-cert.pfx`) in the root of the source directory which sidesteps all the`make` issues and should be easier to test. It should be modestly harder to accidentally leak the secret as well.